app/core/repository.py

```diff
@@ -25,7 +26,7 @@
 DOCUMENT_VALUE_TYPE = Union[MutableMapping, List[MutableMapping]]
 
 
-class IRepository(Protocol):
+class IRepository(ABC):
```

Comment:
Wasn't sure if I should rename IRepository to just Repository.  Happy to if that's more idiomatic.